--- comment.py
+++ comment2.py
@@ -5,10 +5,11 @@

     comment_count = 0
 
-    def __init__(self, content, tags = None):
+    def __init__(self, author, content, tags = None):
         if not self.is_valid_content(content):
             raise ValueError("Invalid comment content.")
         
+        self._author = author
         self._content = content
         self._tags = set()
         if tags is not None:
@@ -53,4 +54,12 @@
 
     @staticmethod
     def is_valid_content(content):
-        return 3 <= len(content) <= 2200
+        return 3 <= len(content) <= 2200
    
+    def like(self, user):
+        if user not in self._liked_by:
+            self._liked_by.append(user)
+
+    def unlike(self, user):
+        if user in self._liked_by:
+            self._liked_by.remove(user)
