--- file1.py
+++ file2.py
@@ -1,6 +1,7 @@
 import re

 from datetime import datetime

-

+from post import Post
+from comment import Comment

 class User:

     user_count = 0
@@ -17,11 +18,18 @@
     def __init__(self, username, email):
         self._username = username
         self._email = email
         self._bio = ""
         self._joined_on = datetime.now()
-        self.posts = []
+        self._posts = []
+        self._comments = []
         self.following = []
         self.followers = []
         self.messages = []
+        self._liked_posts = []
+        self._liked_comments = []
         User.user_count += 1
+
+    @property
+    def posts(self):
+        return self._posts

     @property
     def username(self):
@@ -46,7 +54,7 @@
         return self._joined_on

     def follow(self, user):
-        if user not in self.following:
+        if user is not self and user not in self.following:
             self.following.append(user)
             user.followers.append(self)

@@ -74,3 +82,42 @@
     @classmethod
     def get_user_count(cls):
         return cls.user_count
+
+    def like_post(self, post):
+        if post not in self._liked_posts:
+            self._liked_posts.append(post)
+            post.add_like(self)
+
+    def unlike_post(self, post):
+        if post in self._liked_posts:
+            self._liked_posts.remove(post)
+            post.remove_like(self)
+
+    def like_comment(self, comment):
+        if comment not in self._liked_comments:
+            self._liked_comments.append(comment)
+            comment.add_like(self)
+
+    def unlike_comment(self, comment):
+        if comment in self._liked_comments:
+            self._liked_comments.remove(comment)
+            comment.remove_like(self)
+
+    def add_post(self, post):
+        self._posts.append(post)
+
+    def remove_post(self, post):
+        if post in self._posts:
+            self._posts.remove(post)
+
+    def add_comment(self, comment):
+        self._comments.append(comment)
+
+    def remove_comment(self, comment):
+        if comment in self._comments:
+            self._comments.remove(comment)
